HerPublicWebsite/Models/Questionnaire/SelectLocalAuthorityViewModel.cs

```diff
@@ -1,23 +1,21 @@
 using System.Collections.Generic;
 using System.Linq;
+using GovUkDesignSystem.Attributes.ValidationAttributes;
 using HerPublicWebsite.BusinessLogic.Models;
 
 namespace HerPublicWebsite.Models.Questionnaire;
 
 public class SelectLocalAuthorityViewModel : QuestionFlowViewModel
 {
-    public Dictionary<string, List<LocalAuthorityDetails>> LocalAuthoritiesByInitial { get; set; }
-
-    public SelectLocalAuthorityViewModel()
-    {
-        LocalAuthoritiesByInitial = LocalAuthorityData
-            .LocalAuthorityDetailsByCustodianCode
-            .ToList()
-            .GroupBy(kvp => kvp.Value.Name[0].ToString().ToUpper())
-            .ToDictionary(
-                group => group.Key,
-                group => group.Select(kvp => new LocalAuthorityDetails(kvp.Value.Name, kvp.Key)).ToList());
-    }
+    public Dictionary<string, List<LocalAuthorityDetails>> LocalAuthoritiesByInitial => LocalAuthorityData
+        .LocalAuthorityDetailsByCustodianCode
+        .Where(kvp => string.IsNullOrEmpty(SearchTerm) || kvp.Value.Name.ToLower().Contains(SearchTerm.ToLower()))
+        .ToList()
+        .GroupBy(kvp => kvp.Value.Name[0].ToString().ToUpper())
+        .ToDictionary(
+            group => group.Key,
+            group => group.Select(kvp => new LocalAuthorityDetails(kvp.Value.Name, kvp.Key)).ToList());
+    public string SearchTerm { get; set; }
```

Comment:
I'd prefer members to be above calculated values so it's clearer what is needed for this view model to do it's job